quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.config;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Options for the management interface that handles management endpoints (f.e. health and metrics endpoints)
+ */
+public class ManagementOptions {
+
+    public static final Option<Boolean> LEGACY_OBSERVABILITY_INTERFACE = new OptionBuilder<>("legacy-observability-interface", Boolean.class)
+            .category(OptionCategory.MANAGEMENT)
+            .description("If metrics/health endpoints should be exposed on the main HTTP server (not recommended). If set to true, the management interface is disabled.")
+            .defaultValue(Boolean.FALSE)
+            .buildTime(true)
+            .build();
```

Comment:
Should we mark this as deprecated?

Reply:
I was thinking about it and was a little bit skeptical about introducing a 'NEW DEPRECATED' option, but I don't have strong objections against it. Probably it'd make sense as it's only a temporary option.

Reply:
We're deprecating exposing the metrics/health at the normal HTTP interface. This option is to turn back on this deprecated behaviour and we know we'll remove it in the future. Than IMHO qualifies it to be a deprecated new option. :)